src/app/home/home.component.ts

```diff
@@ -15,9 +15,24 @@ export class HomeComponent implements OnInit {
 
   ngOnInit() {
     this.newsService.getRecentNews().subscribe(
-      data => { this.results = data; },
+      data => {
+        this.setDocumentUrl(data);
+        this.results = data;
+      },
       error => console.log(error)
     );
-    this.hostname = this.api.hostnameEPIC;
+  }
+
+  setDocumentUrl(data) {
+    data.forEach(activity => {
+      if(!activity.documentUrl) {
+        return ;
+      }
+      if (activity.documentUrl.includes('http://www') || activity.documentUrl.length === 0) {
```

Comment:
I am not too sure about what we're trying to do here:
* ```activity.documentUrl.length === 0``` should be already caught by the check at line 28
* we are setting ```activity.documentUrl``` to be...itself?
* it looks like at the very least we could change this to something like
```
if (!activity.documentUrl.includes('http://www'){
  activity.documentUrl = `${this.api.hostnameEPIC }${ activity.documentUrl }`;
}
```

Maybe I am missing/forgetting something here though...

Reply:
Fixed all